app/src/main/java/com/mathewsachin/fategrandautomata/scripts/entrypoints/SupportImageMaker.kt

```diff
@@ -32,78 +30,41 @@ fun getFriendImgPath(Index: Int): File {
+        val isInSupport = isInSupport()
 
-        var supportBound = Region(53 * 2, 0, 143 * 2, 110 * 2)
-        val regionAnchor =
-            ImageLocator.SupportRegionTool
+        // the servant and CE images are further to the right in the friend screen
+        val supportBoundX = if (isInSupport) 53 * 2 else 88 * 2
+        var supportBound = Region(supportBoundX, 0, 143 * 2, 110 * 2)
+        val searchRegion = Region(2100, 0, 370, 1440)
 
-        val searchRegion = Region(2100, 0, 300, 1440)
-        val regionArray = searchRegion.findAll(regionAnchor, supportRegionToolSimilarity)
+        val regionAnchor = ImageLocator.SupportRegionTool
+        // At max two Servant+CE are completely on screen, so only those
+        val regionArray = searchRegion.findAll(regionAnchor, supportRegionToolSimilarity).take(2)
 
         val screenBounds = Region(0, 0, Game.ScriptSize.Width, Game.ScriptSize.Height)
 
-        var i = 0
-
-        for (testRegion in regionArray.map { it.Region }) {
-            // At max two Servant+CE are completely on screen
-            if (i > 1)
-                break
-
-            supportBound = if (isInSupport) {
-                supportBound.copy(Y = testRegion.Y - 70 + 68 * 2)
-            } else {
-                // Assume we are on Friend List
-                supportBound.copy(X = supportBound.X + 10, Y = testRegion.Y + 82)
-            }
+        for ((i, testRegion) in regionArray.map { it.Region }.withIndex()) {
+            // in the friend screen, the "Confirm Support Setup" button is higher
+            val newSupportBoundY = testRegion.Y + (if (isInSupport) 66 else 82)
+            supportBound = supportBound.copy(Y = newSupportBoundY)
 
             if (!screenBounds.contains(supportBound))
                 continue
 
-            val pattern = supportBound.getPattern()
-
-            pattern?.use {
-                val servant = it.crop(Region(0, 0, 125, 44))
-                servant.use {
-                    servant.save(
-                        getServantImgPath(i).absolutePath
-                    )
-                }
-
-                val ce = it.crop(Region(0, 80, pattern.width, 25))
-                ce.use {
-                    ce.save(
-                        getCeImgPath(i).absolutePath
-                    )
-                }
-
-                val friendBound = Region(supportBound.X + pattern.width + 220, supportBound.Y - 95, 400, 110)
-                val friendPattern = friendBound.getPattern()
-
-                friendPattern?.use {
-                    friendPattern.save(
-                        getFriendImgPath(i).absolutePath
-                    )
-                }
+            supportBound.getPattern()?.use {
+                extractServantImage(it, i)
+                extractCeImage(it, i)
+                extractFriendNameImage(supportBound, isInSupport, i)
             }
-
-            ++i
         }
 
-        if (i == 0) {
+        if (regionArray.count() == 0) {
```

Comment:
```suggestion
        if (regionArray.isEmpty()) {
```

after my review comment above, this way seems more Kotlin like.

Otherwise, nice refactoring 👍